cpu/samd5x/cpu.c

```diff
@@ -218,8 +218,8 @@ static void fdpll_init_nolock(uint8_t idx, uint32_t f_cpu, uint8_t flags)
 }
 
 static void fdpll_lock(uint8_t idx) {
-    while (!(OSCCTRL->Dpll[idx].DPLLSTATUS.bit.CLKRDY &&
-             OSCCTRL->Dpll[idx].DPLLSTATUS.bit.LOCK)) {}
+    const uint32_t flags = (OSCCTRL_DPLLSTATUS_CLKRDY | OSCCTRL_DPLLSTATUS_LOCK);
+    while (!((OSCCTRL->Dpll[idx].DPLLSTATUS.reg & flags) == flags)) {}
```

Comment:
This will even be more efficient. I believe normally, the optimizer would combine the two bit-field accesses on the same word to a single memory access. But since those registers are marked as `volatile`, the compiler is not allowed to combine memory accesses.